src/listener/to_listener.rs

```diff
@@ -0,0 +1,124 @@
+use super::{Listener, MultiListener, TcpListener, UnixListener};
+use crate::http::url::Url;
+use async_std::net::ToSocketAddrs;
+
+pub trait ToListener<State: 'static> {
+    fn to_listener(self) -> Box<dyn Listener<State>>;
+}
+
+impl<State: 'static + Send + Sync> ToListener<State> for async_std::path::PathBuf {
+    fn to_listener(self) -> Box<dyn Listener<State>> {
+        Box::new(UnixListener::from_path(self))
+    }
+}
+
+impl<State: 'static + Send + Sync> ToListener<State> for std::path::PathBuf {
+    fn to_listener(self) -> Box<dyn Listener<State>> {
+        Box::new(UnixListener::from_path(self))
+    }
+}
+
+impl<State: 'static + Send + Sync> ToListener<State> for async_std::net::TcpListener {
+    fn to_listener(self) -> Box<dyn Listener<State>> {
+        Box::new(TcpListener::from_listener(self))
+    }
+}
+
+impl<State: 'static + Send + Sync> ToListener<State> for std::net::TcpListener {
+    fn to_listener(self) -> Box<dyn Listener<State>> {
+        Box::new(TcpListener::from_listener(self))
+    }
+}
+
+impl<State: 'static + Send + Sync> ToListener<State> for (&str, u16) {
+    fn to_listener(self) -> Box<dyn Listener<State>> {
+        Box::new(TcpListener::from_addrs(
+            async_std::task::block_on(self.to_socket_addrs())
```

Comment:
There doesn't seem to be a way to make an async_std::net::SocketAddr from a std::net::SocketAddr, which would be nice to be able to avoid block_on here